theano/scan_module/scan_perform.pyx

```diff
@@ -256,9 +254,11 @@ def perform(
     offset = nit_sot_arg_offset + n_nit_sot
     other_args = args[offset:]
     input_storage = fnct.input_storage
-    len_input_storage = len(input_storage)
-    old_input_storage = [None] * len_input_storage
-    old_input_data = [None] * len_input_storage
+    nb_mitmot_in = 0
+    for idx in range(n_mit_mot):
+        nb_mitmot_in += tap_array_len[idx]
+    old_mitmot_input_storage = [None] * nb_mitmot_in
+    old_mitmot_input_data = [None] * nb_mitmot_in
```

Comment:
You might be able to go even faster by making these C arrays instead of python lists.